py/visdom/__init__.py

```diff
@@ -503,6 +507,15 @@ def register_event_handler(self, handler, target):
     def clear_event_handlers(self, target):
         self.event_handlers[target] = []
 
+
+    def get_all_env(self):
+
+        if self.DEFAULT_ENV_PATH is not None:
+            items = [i.replace('.json', '') for i in os.listdir(self.DEFAULT_ENV_PATH) if '.json' in i]
+        else:
+            items = []
+        return items
```

Comment:
This misses the mark - usually `visdom`'s client is run on a different machine than the `visdom` server, and as such the contents of `DEFAULT_ENV_PATH` will usually be empty on a client machine. You should instead be creating an `offline_env_list` or something like that in `__init__` if `offline`, and then in `send` if `offline`, check the `env` that visdom is trying to write to.

Reply:
can you elaborate what you meant by check the "env" that visdom is trying to write to.

Reply:
Check the args of what `visdom` is sending when calling `send` - an env can be included in every one of visdom's functions, like here: https://github.com/fossasia/visdom/blob/06f9cb22fdc4da2cb74f65faf0d6ec64d89e03fc/py/visdom/__init__.py#L884
It ends up being managed as an `eid`: https://github.com/fossasia/visdom/blob/06f9cb22fdc4da2cb74f65faf0d6ec64d89e03fc/py/visdom/__init__.py#L902
And send sets it to the default if none is provided:
https://github.com/fossasia/visdom/blob/master/py/visdom/__init__.py#L667-L668